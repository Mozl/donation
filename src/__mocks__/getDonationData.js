module.exports = {
  getDonationData: () => {
    return Promise.resolve({
      data: [
        {
          amount: 5,
          donorDisplayName: 'Louis Moselhi'
        },
        {
          amount: 10,
          donorDisplayName: 'Elon Musk'
        },
        {
          amount: 1,
          donorDisplayName: 'The Donald'
        },
        {
          amount: 200,
          donorDisplayName: 'Stephen Hawking'
        }
      ]
    });
  }
};
