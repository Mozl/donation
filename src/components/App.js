import React, { Component } from 'react';
import Donation from './Donation';
import styled from 'styled-components';
import MastHead from './MastHead';

const Wrapper = styled.div`
  width: 100%;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  .h1 {
    margin-top: 0.5rem;
    position: absolute;
  }
`;

const DonationWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: []
    };
  }

  getDonationData = () => {
    return fetch(
      `https://api.justgiving.com/beb9b39c/v1/charity/13441/donations`,
      { headers: { 'Content-Type': 'application/json' } }
    )
      .then(res => res.json())
      .then(res => {
        const donationData = res.donations;
        this.setState({ data: donationData });
      });
  };

  componentDidMount() {
    this.getDonationData();
  }

  render() {
    const { data } = this.state;
    return (
      <Wrapper>
        <MastHead />
        <h1>Recent Oxfam Donors</h1>
        <DonationWrapper>
          <Donation data={data} />
        </DonationWrapper>
      </Wrapper>
    );
  }
}

export default App;
