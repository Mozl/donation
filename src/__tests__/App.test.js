import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import getDonationData from '../dataRequest';

const dummyDonationData = [
  { title: 'some-tilte-1', body: 'some-1' },
  { title: 'some-tilte-2', body: 'some-2' },
  { title: 'some-tilte-3', body: 'some-3' }
];
global.fetch = jest.fn(() => Promise.resolve(dummyDonationData));

jest.mock('../dataRequest');

describe('App component', () => {
  describe('when rendered', () => {
    it('should fetch a list of donations', () => {
      getDonationData.mockResolvedValue({ data: {} });
      const getSpy = jest.spyOn(getDonationData, 'getDonationData');
      expect(getSpy).toBeCalled();
    });
  });
});

it('renders correctly', () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});

test('the data is object', () => {
  return getDonationData().then(data => {
    expect(data).toEqual({ data: {} });
  });
});
