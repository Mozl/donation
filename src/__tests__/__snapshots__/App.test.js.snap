// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-ifAKCX cPMSlA"
>
  <div
    className="sc-bwzfXH iKrrve"
  >
    <img
      alt="Just Giving"
      src="https://www.jg-cdn.com/chrome/3e7a4dc9d866fda15331414a9072da97.svg"
    />
    <div
      className="sc-htpNat lhzRTC"
    >
      <button
        className="sc-bxivhb iHLsul"
      >
        About Us
      </button>
    </div>
  </div>
  <h1>
    Recent Oxfam Donors
  </h1>
  <div
    className="sc-EHOje fpqoIO"
  />
</div>
`;
